ranger-http/src/main/java/com/flipkart/ranger/http/servicefinder/HttpNodeDataSource.java

```diff
@@ -23,59 +41,66 @@
+        Preconditions.checkNotNull(config, "client config has not been set for node data");
+        Preconditions.checkNotNull(mapper, "mapper has not been set for node data");
         val httpUrl = new HttpUrl.Builder()
                 .scheme(config.isSecure()
                         ? "https"
                         : "http")
                 .host(config.getHost())
                 .port(config.getPort() == 0
-                      ? defaultPort()
-                      : config.getPort())
+                        ? defaultPort()
+                        : config.getPort())
                 .encodedPath(String.format("/ranger/nodes/v1/%s/%s", service.getNamespace(), service.getServiceName()))
                 .build();
         val request = new Request.Builder()
                 .url(httpUrl)
                 .get()
                 .build();
+
+        List<ServiceNode<T>> serviceNodes = Collections.emptyList();
         try (val response = httpClient.newCall(request).execute()) {
             if (response.isSuccessful()) {
                 try (final ResponseBody body = response.body()) {
                     if (null == body) {
                         log.warn("HTTP call to {} returned empty body", httpUrl);
-                    }
-                    else {
+                    } else {
                         final byte[] bytes = body.bytes();
-                        return Optional.of(FinderUtils.filterValidNodes(
-                                service,
-                                deserializer.deserialize(bytes),
-                                healthcheckZombieCheckThresholdTime(service)));
+                        val serviceNodesResponse = deserializer.deserialize(bytes);
+                        if(serviceNodesResponse.isSuccess()){
+                            serviceNodes = FinderUtils.filterValidNodes(
+                                    service,
+                                    serviceNodesResponse.getData(),
+                                    healthcheckZombieCheckThresholdTime(service));
+                        }else{
+                            log.warn("Http call to {} returned a failure response with error {}", httpUrl, serviceNodesResponse.getError());
+                        }
                     }
                 }
+            } else {
+                log.warn("HTTP call to {} returned: {}", httpUrl, response.code());
             }
-            else {
-                log.warn("HTTP call to {} returned: {}", httpUrl.toString(), response.code());
-            }
-        }
-        catch (IOException e) {
-            Exceptions.illegalState("Error fetching data from server: " + httpUrl, e);
+        } catch (IOException e) {
+            log.error("Error getting service data from the http endPoint: ", e);
         }
-        log.error("No data returned from server: " + httpUrl);
-        return Optional.empty();
+        return Optional.of(serviceNodes);
```

Comment:
Return Collections.emptyList() from here

Reply:
This was changed as part of moving all lists to not-optional.